playwright-e2e/pages/singular/navbar.ts

```diff
@@ -38,7 +34,12 @@ export async function visitHomePage(page: Page): Promise<Response | null> {
 }
 
 export async function signMeUp(page: Page): Promise<void> {
-  return await page.locator(NavSelectors.SignMeUp).click();
+  const progressSpinner = 'mat-spinner[role="progressbar"]';
+  await Promise.all([
+    page.locator(NavSelectors.SignMeUp).click(),
+    page.locator(progressSpinner).waitFor({ state: 'visible' })
+  ]);
+  await page.locator(progressSpinner).waitFor({ state: 'detached', timeout: 30 * 1000 });
```

Comment:
On CI VM, there was one time when the progress spinner took longer than 10 seconds, due to slower network speed on CI.